js/reference.js

```diff
@@ -4350,75 +4428,266 @@
      *
      * If the ReferenceColumn is not provided, then the FacetColumn is for reference
      *
-     * @param {ReferenceColumn=} refColumn reference column
+     * @param {ERMrest.Reference} reference the reference that this FacetColumn blongs to.
+     * @param {int} index The index of this FacetColumn in the list of facetColumns
+     * @param {obj|string} json the filter object
      * @constructor
      */
-    function FacetColumn (reference, refColumn, index) {
-        verify(!refColumn.isAsset && !refColumn.isKey, "Facet Column cannot be asset or key pseudo-column.");
-        verify(!refColumn.isForeignKey || refColumn.foreignKey.simple, "Facet Column does not support composite foreign keys.");
-
+    function FacetColumn (reference, index, column, json, filters) {
+        
         /**
-         * The {@link ERMrest.ReferenceColumn} that this facet belongs to. If it is
-         * undefined, that means this facet belongs to the table and not column.
-         * @type {ERMrest.ReferenceColumn}
+         * The column object that the filters are based on
+         * @type {ERMrest.Column}
+         */
+        this._column = column;
+        
+        /**
+         * [reference description]
+         * @type {ERMrest.Reference}
          */
-        this.column = refColumn;
-
         this.reference = reference;
-
+        
+        /**
+         * The index of facetColumn in the list of facetColumns
+         * NOTE: Might not be needed
```

Comment:
This is something that I have used in the implementation of `FacetColumn`. The Note is just for when I get to refactor it, we might not need to pass this index around